pkg/repoowners/repoowners.go

```diff
@@ -304,22 +318,50 @@ func loadAliasesFrom(baseDir string, log *logrus.Entry) RepoAliases {
 		log.WithError(err).Warnf("Failed to read alias file %q. Using empty alias map.", path)
 		return nil
 	}
-	config := &struct {
-		Data map[string][]string `json:"aliases,omitempty"`
-	}{}
-	if err := yaml.Unmarshal(b, config); err != nil {
-		log.WithError(err).Errorf("Failed to unmarshal aliases from %q. Using empty alias map.", path)
-		return nil
-	}
-
 	result := make(RepoAliases)
-	for alias, expanded := range config.Data {
-		result[scmprovider.NormLogin(alias)] = normLogins(expanded)
+	err = c.parseAliases(result, b, log, org)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to unmarshal aliases from %q.", path)
 	}
 	log.Infof("Loaded %d aliases from %q.", len(result), path)
 	return result
 }
 
+func (c *Client) parseAliases(result RepoAliases, b []byte, log *logrus.Entry, org string) error {
+	config := &OwnerAliases{}
+	err := yaml.Unmarshal(b, config)
+	if err != nil {
+		return err
+	}
+	if config.ForeignAliases != nil {
+		for _, r := range config.ForeignAliases {
+			if r.Ref == "" {
+				r.Ref = "HEAD"
+			}
+			if r.Org == "" {
+				r.Org = org
+			}
+			b, err = c.spc.GetFile(r.Org, r.Repo, aliasesFileName, r.Ref)
+			if err == nil {
```

Comment:
Return error using err != nil? This error is not being handled, same with `c.parseAliases`, we should return those errors before we proceed in the code imo ...

Reply:
The error is returned in the end of the function. But we don't want to abort the processing because of an error.
The only drawback with the current code is that if there are several errors only the last will be reported. But I think that is OK.

Reply:
can we try to aggregate the errors then? we dont want to lose the errors, else the end user will have a hard time knowing the exact issue.

Reply:
We could. But since this can only happen in the case several repos are specified and there are failures in accessing several of them it is really an edge case that I don't know if it's really worth adding complexity to solve.

Reply:
I also wonder if it will get flagged by errcheck when linting is enabled on lighthouse in the next few weeks 🤔 
@rawlingsj wdyt?

Reply:
I think it's fine for now, I will see what happens when we add linting to lighthouse.

Reply:
I do see that we handle the error from find at other places, approved, but putting a hold for @rawlingsj's review.

Reply:
yeah that all sounds good to me